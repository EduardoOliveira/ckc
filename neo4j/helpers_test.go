package neo4j

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestMergeCountryCypher(t *testing.T) {
	t.Run("test simple", func(t *testing.T) {
		t.Parallel()
		key, cypher, props := MergeCountryCypher("pt", "Portugal", map[string]any{
			"country_code":  "PT",
			"has_coastline": true,
		})
		snaps.MatchSnapshot(t, key, cypher, props)
	})
}

func TestMergeCityWithCountryCypher(t *testing.T) {
	t.Run("test simple", func(t *testing.T) {
		t.Parallel()
		key, cypher, props := MergeCityWithCountryCypher(1, "Lisbon", "Portugal", map[string]any{
			"population": 500000,
			"area":       100,
		})
		snaps.MatchSnapshot(t, key, cypher, props)
	})
}

func TestMatchIPAddressCypher(t *testing.T) {
	t.Run("test simple", func(t *testing.T) {
		t.Parallel()
		key, cypher := MergeIPAddressCypher(1, "127.0.0.1")
		snaps.MatchSnapshot(t, key, cypher)
	})
}
