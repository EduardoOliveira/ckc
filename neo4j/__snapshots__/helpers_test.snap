
[TestMergeCountryCypher/test_simple - 1]
pt
MERGE (pt:Country {name: $pt_name})
SET pt.country_code = $pt_country_code, pt.has_coastline = $pt_has_coastline
WITH pt 

map[string]interface {}{
    "pt_country_code":  "PT",
    "pt_has_coastline": bool(true),
    "pt_name":          "Portugal",
}
---

[TestMergeCityWithCountryCypher/test_simple - 1]
city_1
MERGE (city_1:City {name: $city_1_name})
SET city_1.area = $city_1_area, city_1.population = $city_1_population
WITH city_1 

map[string]interface {}{
    "city_1_area":       int(100),
    "city_1_name":       "Lisbon",
    "city_1_population": int(500000),
}
---

[TestMatchIPAddressCypher/test_simple - 1]
ip_1
MERGE (ip_1:IPAddress {address: "127.0.0.1"})
WITH ip_1
---
